crawl-ref/source/startup.cc

```diff
@@ -363,6 +363,7 @@ static void _post_init(bool newc)
     update_vision_range();
     init_exclusion_los();
     ash_check_bondage();
+    check_if_everything_is_identified();
```

Comment:
as noted above, if this were implemented with a player prop, this call would be unnecessary.

Reply:
> Good change, but I have some comments on the implementation. My recommendation is either: (i) ensure that the `static` has a real default initialization, or (ii) implement this using a prop on the player rather than a `static`. My preference is for (ii), though (i) is closer to the current patch.

Regarding (i), I suppose [Zero Initialisation](https://en.cppreference.com/w/cpp/language/zero_initialization) is more a C thing, too. In this case, _is_everything_identified is initially false.

I've changed it to use (ii), though.

> This function could return immediately if its condition (however implemented) is already marked as satisfied, as long as it has a default initialization.

I've changed it to skip this check when called by _identify_last_item(). Other than in wizard mode, I don't think any other function can identify a scroll or potion once scrolls of identify are redundant.

> as noted above, if this were implemented with a player prop, this call would be unnecessary.

As a new version could contain different items which need to be identified, I've changed it to only calculates it for a new game or when loading a game with a different version string.